### DISCUSSION TITLE
IPSW location for macOS guest

### DISCUSSION THREAD

**Discussion by ryansch** [utmapp/UTM#4097] [Ideas]
I expected that after I downloaded the IPSW the first time, UTM would prompt me to use it or at least open the file browser to the location of the cache.

For those finding this via a search, the cache location is: `~/Library/Containers/com.utmapp.UTM/Data/Library/Caches`

**Comment by evangeloszotos** [comment]
On my 2021 MacBook the cache for IPSW files location is: 

` ~/Library/Containers/UTM/Data/Library/Caches`

I would at leats expect it in the docs. 
I spent an hour searching for it, even after I found your post.

**Reply by hotdang-ca** [reply]
I still found it in `~/Library/Containers/com.utmapp.UTM`. 

Thanks! Glad to get that 12GB back.

**Reply by keeely** [reply]
All that needs to happen IMHO is that the browse button opens ~/Library/Containers/com.utmapp.UTM/Data/Library/Caches, and then the user can take it from there.

**Comment by coolaj86** [comment]
# Exact Location

The exact location for me was this:

```text
~/Library/Containers/com.utmapp.UTM/Data/Library/Caches/UniversalMac_13.5.2_22G91_Restore.ipsw
```

It exists only after the entire download has finished after creating the template.

It's deleted immediately after first boot.

The current macOS is 13.6.1 (2023), but it downloaded 13.5.2 (2022). That stinks because it means it will immediately download a huge update that causes the disk size to pretty much double and not shrink back down again.

# To Save for Later

I saved it for later use by

```sh
mv ~/Library/Containers/com.utmapp.UTM/Data/Library/Caches/UniversalMac_13.5.2_22G91_Restore.ipsw ~/Downloads/
ln -s ~/Downloads/UniversalMac_13.5.2_22G91_Restore.ipsw ~/Library/Containers/com.utmapp.UTM/Data/Library/Caches/
```

# To Get Latest Releases

Doing a quick search I found much more recent versions at:
- <https://mesu.apple.com/assets/macos/com_apple_macOSIPSW/com_apple_macOSIPSW.xml>
   ```sh
   curl https://mesu.apple.com/assets/macos/com_apple_macOSIPSW/com_apple_macOSIPSW.xml |
       sed 's:[<>]:\n:g' |
       grep ipsw |
       grep UniversalMac_ |
       sort -u
   ```
- <https://ipsw.me/>
- <https://osxdaily.com/gsearch/?q=macos+13.6>

For example:
- `UniversalMac_13.6_22G120_Restore.ipsw`
- `UniversalMac_14.1.2_23B92_Restore.ipsw`

I haven't tested these yet, but since I've run the updates in UTM and that's worked, I assume these work as well.

# Patches

I found a list of patches at <https://osxdaily.com/gsearch/?q=macos+13.6>.

For example:
- 13.6.1: <https://swcdn.apple.com/content/downloads/14/43/042-82952-A_GEJ3YDGEVG/cxed0qvmvtdvuix3xskr6zz1o4nc8fwry0/InstallAssistant.pkg>

Yet the "patch" is _obscenely_ large - **11.1 GB**!

I can put that in a _Shared Folder_ to save on the download size, but it will still bloat the disk when it overwrites the whole OS.
